src/osp/Active/ActiveScene.h

```diff
@@ -228,32 +237,31 @@ class ActiveScene
 
 // move these to another file eventually
 
-template<class T, typename... Args>
-void ActiveScene::system_machine_add(std::string const& name,
-                                             Args &&... args)
+
+template<class SYSMACH_T, typename... ARGS_T>
+void ActiveScene::system_machine_create(ARGS_T &&... args)
 {
-    m_sysMachines.emplace(name, std::make_unique<T>(*this, args...));
+    system_machine_add(SYSMACH_T::smc_name,
+                       std::make_unique<SYSMACH_T>(*this, args...));
 }
 
-template<class T, typename... Args>
-T& ActiveScene::dynamic_system_add(std::string const& name,
-                                             Args &&... args)
+
+template<class DYNSYS_T, typename... ARGS_T>
+DYNSYS_T& ActiveScene::dynamic_system_add(ARGS_T &&... args)
 {
-    auto ptr = std::make_unique<T>(*this, args...);
-    T &refReturn = *(ptr.get());
+    auto ptr = std::make_unique<DYNSYS_T>(*this, args...);
```

Comment:
When passing template parameter packs into a function as &&'s, you have to make sure you use std::forward<ARGS_T>(args)...

if you don't, the move constructor won't ever be called, even when you expect it to.